src/main/java/pl/zaradny/springApp/domain/ProductFacadeImpl.java

```diff
@@ -0,0 +1,61 @@
+package pl.zaradny.springApp.domain;
+
+import org.springframework.stereotype.Component;
+import pl.zaradny.springApp.infrastructure.ProductRepository;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Component
+public class ProductFacadeImpl implements ProductFacade {
+
+    private final ProductRepository productRepository;
+
+    ProductFacadeImpl(ProductRepository productRepository){
+        this.productRepository = productRepository;
+    }
+
+
+    @Override
+    public ProductResponseDto findById(String id){
+        Product product = productRepository.findById(id);
+        return new ProductResponseDto(product.getId(), product.getName());
+    }
+
+    @Override
+    public ProductResponseDto create(ProductRequestDto productRequest) {
+
+        if(!productRequest.isValid()){
+            throw new RuntimeException("Product can not be empty!");
+        }
+
+        String id = UUID.randomUUID().toString();
+        LocalDateTime createdAt = LocalDateTime.now();
+        Product product = new Product(id, productRequest.getName(), createdAt);
+
+        productRepository.save(product);
+        return new ProductResponseDto(product.getId(), product.getName());
+
+    }
+
+    @Override
+    public ProductResponseDto deleteById(String id) {
+        Product product = productRepository.findById(id);
+        productRepository.deleteById(id);
+        return new ProductResponseDto(product.getId(), product.getName());
+    }
+
+    @Override
+    public ProductResponseDto updateById(String id, ProductRequestDto productRequestDto) {
+
+        if(!productRequestDto.isValid()){
+            throw new RuntimeException("Product can not be empty!");
```

Comment:
Good job. Przy aktualizacji walidacja jak najbardziej także powinna zachodzić :) Pytanie tylko, czy to, że _productRequestDto_ został określony jako niepoprawny wprost mówi nam o tym, że _Product is empty_? Dzisiaj walidacja faktycznie sprawdza nam tylko nazwę produktu, ale jeżeli jutro dodamy nowe pole do walidacji to będzie trzeba poprawić także to miejsce. Masz może jakiś pomysł jak można by to ograć?